plugin/arcconf/arcconf.py

```diff
@@ -0,0 +1,698 @@
+
+    @_handle_errors
+    def capabilities(self, system, flags=Client.FLAG_RSVD):
+        cur_lsm_syss = self.systems()
+        if system.id not in list(s.id for s in cur_lsm_syss):
+            raise LsmError(
+                ErrorNumber.NOT_FOUND_SYSTEM,
+                "System not found")
+        cap = Capabilities()
+        cap.set(Capabilities.DISKS)
+        cap.set(Capabilities.SYS_FW_VERSION_GET)
+        cap.set(Capabilities.SYS_MODE_GET)
+        cap.set(Capabilities.DISK_LOCATION)
+        cap.set(Capabilities.DISK_VPD83_GET)
+        if system.mode != System.MODE_HBA:
+            cap.set(Capabilities.VOLUMES)
+            cap.set(Capabilities.VOLUME_RAID_CREATE)
+            cap.set(Capabilities.VOLUME_DELETE)
+
+        return cap
+
+    def _arcconf_exec(self, arcconf_cmds, flag_force=False):
+        arcconf_cmds.insert(0, self._arcconf_bin)
+        if flag_force:
+            arcconf_cmds.append('noprompt')
+        try:
+            output = cmd_exec(arcconf_cmds)
+        except OSError as os_error:
+            if os_error.errno == errno.ENOENT:
+                raise LsmError(
+                    ErrorNumber.INVALID_ARGUMENT,
+                    "arcconf binary '%s' does not exist." %
+                    self._arcconf_bin)
+            else:
+                raise
+
+        return output
+
+    def _get_arcconf_controllers_count(self):
+        total_cntrls = 0
+        try:
+            test = self._arcconf_exec(['list'])
+            output_lines = [l for l in test.split("\n")]
+            first_line = output_lines[0]
+            tmp = first_line.split(":")
+
+            if tmp[0].strip() == 'Controllers found':
+                total_cntrls = int(tmp[1].strip())
+        except ExecError:
+            raise LsmError(
+                ErrorNumber.NOT_FOUND_SYSTEM,
+                "Failed to get controller count")
+
+        return total_cntrls
+
+    def _get_detail_info_list(self):
+        detail_info_list = []
+        ctrl_count = self._get_arcconf_controllers_count()
+
+        for ctrl_no in range(ctrl_count):
+            output = self._arcconf_exec(["getconfigJSON", str(ctrl_no + 1)])
+            detail_info = json.loads(output.split("\n")[1])
+            detail_info_list.append(detail_info)
+
+        return detail_info_list
+
+    @_handle_errors
+    def systems(self, flags=0):
+        """
+        Depend on command:
+            arcconf getconfig <cntrl_no>
+        """
+        rc_lsm_syss = []
+        total_cntrl = self._get_arcconf_controllers_count()
+        getconfig_cntrls_info = self._get_detail_info_list()
+
+        for cntrl in range(total_cntrl):
+            ctrl_info = getconfig_cntrls_info[cntrl]['Controller']
+            ctrl_name = "%s %s" % (ctrl_info['deviceVendor'],
+                                   ctrl_info['deviceName'])
+            sys_id = str(cntrl + 1)
+            status = \
+                getconfig_cntrls_info[cntrl]['Controller']['controllerStatus']
```

Comment:
You should use `ctrl_info` here, in stead of `getconfig_cntrls_info[cntrl]['Controller']`.

There are many lines like this below also.